winlogbeat/checkpoint/checkpoint.go

```diff
@@ -89,6 +89,12 @@ func NewCheckpoint(file string, maxUpdates int, interval time.Duration) (*Checkp
 		}
 	}
 
+	// Write the state file to verify we have have permissions.
+	err = c.flush()
```

Comment:
If no state was read from a previous run, then the `flush `will write out a mostly empty file containing just a timestamp.